File: website/src/features/inventory/domain/entities/InventoryMetadata.ts
/**
 * Inventory Metadata Entities
 * Types for inventory dropdown data from get_inventory_metadata RPC
 */

export interface Category {
  category_id: string;
  category_name: string;
  description: string;
  is_active: boolean;
  product_count: number;
  can_have_subcategory: boolean;
}

export interface Brand {
  brand_id: string;
  brand_name: string;
  description: string;
  is_active: boolean;
  product_count: number;
}

export interface ProductType {
  value: string;
  label: string;
  description: string;
  product_count: number;
}

export interface Unit {
  value: string;
  label: string;
  symbol: string;
  product_count: number;
}

export interface Currency {
  code: string;
  name: string;
  symbol: string;
}

export interface StockStatusLevel {
  level: string;
  label: string;
  color: string;
  icon: string;
}

export interface ValidationRules {
  sku_pattern: string;
  barcode_pattern: string;
  max_stock_required: boolean;
  min_price_required: boolean;
}

export interface AllowCustomValues {
  units: boolean;
  brands: boolean;
  categories: boolean;
}

export interface StoreInfo {
  store_id: string | null;
  store_code: string | null;
  store_name: string | null;
}

export interface Stats {
  total_categories: number;
  total_brands: number;
  total_products: number;
  active_products: number;
  inactive_products: number;
}

export interface InventoryMetadata {
  company_id: string;
  categories: Category[];
  brands: Brand[];
  product_types: ProductType[];
  units: Unit[];
  currency: Currency;
  stock_status_levels: StockStatusLevel[];
  validation_rules: ValidationRules;
  allow_custom_values: AllowCustomValues;
  store_info: StoreInfo;
  stats: Stats;
  last_updated: string;
}

export interface InventoryMetadataResponse {
  success: boolean;
  data: InventoryMetadata;
}

File: myFinance_improved_V2/lib/features/sale_product/domain/entities/inventory_metadata.dart
/// Inventory metadata entity
class InventoryMetadata {
  final List<BrandMetadata> brands;
  final List<CategoryMetadata> categories;
  final List<ProductTypeMetadata> productTypes;
  final List<UnitMetadata> units;
  final StatsMetadata stats;
  final CurrencyMetadata? currency;
  final StoreInfoMetadata? storeInfo;

  const InventoryMetadata({
    required this.brands,
    required this.categories,
    required this.productTypes,
    required this.units,
    required this.stats,
    this.currency,
    this.storeInfo,
  });

  factory InventoryMetadata.empty() => const InventoryMetadata(
        brands: [],
        categories: [],
        productTypes: [],
        units: [],
        stats: StatsMetadata(
          totalProducts: 0,
          activeProducts: 0,
          inactiveProducts: 0,
          totalCategories: 0,
          totalBrands: 0,
        ),
      );
}

class BrandMetadata {
  final String brandId;
  final String brandName;
  final String? description;
  final bool isActive;
  final int productCount;

  const BrandMetadata({
    required this.brandId,
    required this.brandName,
    this.description,
    required this.isActive,
    required this.productCount,
  });
}

class CategoryMetadata {
  final String categoryId;
  final String categoryName;
  final String? description;
  final bool isActive;
  final int productCount;
  final bool canHaveSubcategory;

  const CategoryMetadata({
    required this.categoryId,
    required this.categoryName,
    this.description,
    required this.isActive,
    required this.productCount,
    required this.canHaveSubcategory,
  });
}

class ProductTypeMetadata {
  final String value;
  final String label;
  final String? description;
  final int productCount;

  const ProductTypeMetadata({
    required this.value,
    required this.label,
    this.description,
    required this.productCount,
  });
}

class UnitMetadata {
  final String value;
  final String label;
  final String? symbol;
  final int productCount;

  const UnitMetadata({
    required this.value,
    required this.label,
    this.symbol,
    required this.productCount,
  });
}

class StatsMetadata {
  final int totalProducts;
  final int activeProducts;
  final int inactiveProducts;
  final int totalCategories;
  final int totalBrands;

  const StatsMetadata({
    required this.totalProducts,
    required this.activeProducts,
    required this.inactiveProducts,
    required this.totalCategories,
    required this.totalBrands,
  });
}

class CurrencyMetadata {
  final String? code;
  final String? name;
  final String? symbol;

  const CurrencyMetadata({
    this.code,
    this.name,
    this.symbol,
  });
}

class StoreInfoMetadata {
  final String storeId;
  final String? storeName;
  final String? storeCode;

  const StoreInfoMetadata({
    required this.storeId,
    this.storeName,
    this.storeCode,
  });
}
